src/finch/compiled.py

```diff
@@ -0,0 +1,35 @@
+from functools import wraps
+
+from .julia import jl
+from .tensor import Tensor
+
+
+def compiled(func):
+    @wraps(func)
+    def wrapper_func(*args, **kwargs):
+        for arg in args:
+            if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LogicTensor):
+                arg._obj = jl.Finch.LogicTensor(arg._obj)
+
+        result = func(*args, **kwargs)
+        result_tensor = Tensor(jl.Finch.compute(result._obj))
+
+        for arg in args:
+            arg._obj = arg._obj.data
+
+        return result_tensor
+
+    return wrapper_func
+
+
+def LogicTensor(tensor: Tensor):
```

Comment:
Do we want a LogicTensor in Python, or do we just want the existing Tensor to hold a Julia LogicTensor?